test/controllers/api/v2/repositories_controller_test.rb

```diff
@@ -543,6 +541,15 @@ def test_show_protected
       end
     end
 
+    def test_show_protected_specific_instance
+      allowed_perms = [{:name => @read_permission, :search => "name=\"#{@repository.product.name}\"" }]
+      denied_perms = [{:name => @read_permission, :search => "name=\"#{@redhat_repository.product.name}\"" }]
+
+      assert_protected_action(:show, allowed_perms, denied_perms, expect_404: true) do
```

Comment:
what do you think about having a shortcut method  'assert_protected_object'  , which just calls assert_protected_action, passing expect_404?

Reply:
added